File: Algorithms/Java/Pascal's Triangle.java
public class Solution {
    public List<List<Integer>> generate(int numRows) {
		List<List<Integer>> ans = new ArrayList<List<Integer>>();
		List<Integer> s = new ArrayList<Integer>();
		if (numRows != 0) {
			s.add(1);
			ans.add(s);
		}
		for (int i = 1; i < numRows; i++) {
			List<Integer> prev = ans.get(i - 1);
			List<Integer> curr = new ArrayList<Integer>();
			for (int j = 0; j <= i; j++) {
				if (j == 0 || j == i )
					curr.add(1);
				else
					curr.add(prev.get(j - 1) + prev.get(j));
			}
			ans.add(curr);
		}
		return ans;
	}

}

File: Algorithms/Python/Pascal's-Triangle.py
class Solution(object):

    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        ans = []
        for i in range(numRows):
            temp = [1] * (i + 1)
            for j in range(i + 1):
                if j == 0 or j == i:
                    pass
                else:
                    temp[j] = ans[i - 1][j - 1] + ans[i - 1][j]
            ans.append(temp)
        return ans
